### PR TITLE
[@types/react] Remove `placeholder` from HTMLAttributes

### LINKED ISSUE #26736: [@types/react] `placeholder` attribute should not be on HTMLAttributes
- [x] I tried using the `@types/react` package and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @epzilla @mhegazy @JoshuaKGoldberg

I realize now you may not have seen [my previous comment](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/22933#issuecomment-395091847) because it was on a closed issue.

TLDR: `placeholder` is not a valid attribute on all DOM elements (only input and textarea), therefore #22933 should have been a WONTFIX, and we should revert #24539 because alternative solutions exist, unless [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) are not an option.

### PR DESCRIPTION
**Pull request by emillaine**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/placeholder
  - https://html.spec.whatwg.org/multipage/dom.html#global-attributes
  - https://html.spec.whatwg.org/multipage/input.html#the-placeholder-attribute
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

Additional context:

This attribute was originally proposed in #22933, and merged in #24539, without much community discussion.

Soon #26736 was opened to propose reverting this change, with more upvotes and approving comments than the original issue. But it was closed as stale.

To summarize, `placeholder` is only supported on `input` and `textarea` in pure HTML. For custom `contenteditable` elements, it's better practice to use a custom `data-placeholder` attribute (to be more future-proof regarding future HTML spec updates).

Another real world problem that the original change introduced is that prop autocompletion suggests and allows the `placeholder` prop for all elements, as well as custom React components that inherit the HTML props for their root element, even if a placeholder is not supported.

Fixes #26736.
Reverts #22933, #24539.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react@latest/))
was missing the following properties:
1. unstable_act
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 96e79e8a1ac12de17ce72cdad608cad3ffed4f86
</p>

**Comment by cssinate** [comment]
> Hey @emlai,
> 
> 😒 Your PR doesn't modify any tests, so it's hard to know what's being fixed, and your changes might regress in the future. Please consider [adding tests](https://github.com/DefinitelyTyped/DefinitelyTyped#user-content-test-editing-an-existing-package) to cover the change you're making. Including tests allows this PR to be merged by yourself and the owners of this module.
> 
> _**This can potentially save days of time for you!**_

Indeed - why not amend the PR to add tests to confirm that `placeholder` throws an error on a few places we distinctly don't want it?

**Review by eps1lon — CHANGES_REQUESTED** [review]
Thank you for the change and the detailed context.

 Could you also add a test with 
```tsx
// @ts-expect-error
<div contenteditable placeholder="foo"
```
? I suspect we otherwise re-introduce it in the future with context lost.

Also make sure it's ported to `ts5.0/`

**Comment by emlai** [comment]
Thanks, I'll add a test once I have the time. Sorry, this fix is not a huge priority for me at the moment.

In the meantime, if someone stumbles on this, and has a moment to add the test, feel free to go ahead! :)

**Review by eps1lon — APPROVED** [review]

**Comment by eps1lon** [comment]
Ready to merge
